sw/device/lib/dif/dif_rv_core_ibex.h

```diff
@@ -0,0 +1,122 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RV_CORE_IBEX_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_RV_CORE_IBEX_H_
+
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_base.h"
+
+#include "sw/device/lib/dif/autogen/dif_rv_core_ibex_autogen.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Address translation slot.
+ */
+typedef enum dif_rv_core_ibex_addr_translation_slot {
+  kDifRvCoreIbexAddrTranslationSlot_0,
+  kDifRvCoreIbexAddrTranslationSlot_1,
+  kDifRvCoreIbexAddrTranslationSlotCount,
+} dif_rv_core_ibex_addr_translation_slot_t;
+
+/**
+ * Address tranlation matching region.
+ *
+ * The value programmed is done at power-of-2 alignment. For example, if the
+ * intended matching region is 0x8000_0000 to 0x8000_FFFF, the value
+ * programmed is 0x8000_7FFF.
+ *
+ * The value programmed can be determined from the translation granule. Assume
+ * the user wishes to translate a specific 64KB block to a different address:
+ * 64KB has a hex value of 0x10000. Subtract 1 from this value and then right
+ * shift by one to obtain 0x7FFF. This value is then logically OR'd with the
+ * upper address bits that would select which 64KB to translate.
+ */
+typedef struct dif_rv_core_ibex_addr_translation_pair {
+  /**
+   * Matching address (Virtual address).
+   * When an incoming transaction matches the matching
+   * region, it is redirected to the new address. If a transaction does not
+   * match, then it is directly passed through.
+   */
+  uintptr_t matching_addr;
+
+  /**
+   * Remap address (Physical address).
+   * The region where the matched transtaction will be
+   * redirected to.
+   */
+  uintptr_t remap_addr;
+
+  /**
+   * Address region size.
+   */
+  size_t size;
+} dif_rv_core_ibex_addr_translation_pair_t;
+
+/**
+ * Addresses translation region.
+ */
+typedef struct dif_rv_core_ibex_addr_translation_region {
+  /**
+   * Region representing the instruction bus.
+   */
+  dif_rv_core_ibex_addr_translation_pair_t ibus;
+
+  /**
+   * Region representing the data bus.
+   */
+  dif_rv_core_ibex_addr_translation_pair_t dbus;
+} dif_rv_core_ibex_addr_translation_region_t;
+
+/**
+ * Configure the instruction and data bus in the address translation `slot`.
+ *
+ * @param rv_core_ibex Handle.
+ * @param slot   Slot to be used.
+ * @param region Dbus and Ibus addresses.
+ * @return dif_result_t
```

Comment:
nit (for consistency with other DIFs):
```suggestion
 * @return The result of the operation.
```

applies throughout

Reply:
Done